src/app/common/focus.ts

```diff
@@ -12,8 +12,10 @@ export class AutoFocus {
     }    
 
     ngOnInit() {
-        this._renderer.invokeElementMethod(this._el.nativeElement, 'focus', []);
-
+        setTimeout(() => {
+            this._renderer.invokeElementMethod(this._el.nativeElement, 'focus', []);
+        }, 1);
```

Comment:
This is simply asynchronous call, which is common pattern(recommended) to call focus method from this kind of directive. I confirmed this works and there is no additional delay time because of this change.